vllm-tensorizer/Dockerfile

```diff
@@ -53,12 +68,20 @@ WORKDIR /workspace
 
 RUN apt-get -qq update && apt-get install -y --no-install-recommends curl && apt-get clean
 
-RUN pip3 install "transformers == 4.36.0" "fschat[model_worker] == 0.2.30" "triton == 2.1.0"
+RUN --mount=type=bind,from=freezer,target=/tmp/frozen \
+    /tmp/frozen/freeze.sh torch torchaudio torchvision xformers > /tmp/constraints.txt
+
+RUN python3 -m pip install \
+      "fschat[model_worker] == 0.2.30" "triton == 2.1.0" \
+      -c /tmp/constraints.txt
 
-# Install the vllm wheel from the vllm-builder layer
 RUN --mount=type=bind,from=vllm-builder,source=/wheels,target=/tmp/wheels \
-    python3 -m pip install --no-cache-dir /tmp/wheels/*.whl
+    python3 -m pip install --no-cache-dir /tmp/wheels/*.whl -c /tmp/constraints.txt && \
+    rm /tmp/constraints.txt
+
+# `dive` shows that the `.cache` folder still exists in the final image
+RUN rm -rf /root/.cache/pip
```

Comment:
Running `rm -rf` here will not decrease the final image size because container images are copy-on-write (files are only marked as removed, but still take up space). It most likely exists because another `pip` step is missing `--no-cache-dir`, and needs to be fixed there.

```suggestion
```